assets/js/components/Dashboard/PaMessagesPage.tsx

```diff
@@ -1,29 +1,84 @@
-import React, { ComponentType, useState, useEffect } from "react";
-import { Container, Row, Col, FormCheck } from "react-bootstrap";
+import React, { ComponentType, useState, useMemo, useEffect } from "react";
+import {
+  Container,
+  Row,
+  Col,
+  FormCheck,
+  ButtonGroup,
+  Button,
+} from "react-bootstrap";
 import { PlusCircleFill } from "react-bootstrap-icons";
-import { fetchPaMessages } from "Utils/api";
 import { PaMessage } from "Models/pa_message";
-import { Link } from "react-router-dom";
+import { Link, useSearchParams } from "react-router-dom";
+import cx from "classnames";
+import useSWR from "swr";
+
+type State = "active" | "future" | "past";
```

Comment:
`State` almost seems like too generic a term here considering it could refer to the overall "state" of the page/component. idk if I'd go so far as to change the name of the API query param, but maybe in the context of this frontend code it could be more clearly-named, like "time filter". (Though maybe once we have more filters and presumably those are defined together in some way, it will become more clear that this is a "state filter".)

Reply:
Yeah I started with `stateFilter`. I'll put it back that way. You're right, this is a bit too generic.